domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressListenerImpl.kt

```diff
@@ -0,0 +1,27 @@
+package org.oppia.android.domain.exploration
+
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.domain.oppialogger.analytics.ApplicationLifecycleListener
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Implementation of [ExplorationProgressListener] */
+@Singleton
+class ExplorationProgressListenerImpl @Inject constructor(
+  private val explorationActiveTimeController: ExplorationActiveTimeController
+) : ExplorationProgressListener, ApplicationLifecycleListener {
+  override fun onExplorationSessionStarted() {
+    explorationActiveTimeController.setExplorationSessionStarted()
+  }
+
+  override fun onExplorationSessionPaused(profileId: ProfileId, topicId: String) {
+    explorationActiveTimeController.setExplorationSessionPaused(profileId, topicId)
+  }
+
+  override fun onAppInForeground() {
+    explorationActiveTimeController.setExplorationSessionStarted()
```

Comment:
Isn't this only true if there's an ongoing session? I'd also expect onAppInBackground to play a role in 'pausing' logic.

I think either the listener needs to determine session state based on its various signals & pass this along to the controller, or it needs to forward its received values to the controller and let it perform the determination.

Also a separate thought: you could alternatively do away with this class entirely and just have the controller implement the listeners & bind to the corresponding listener sets.

Reply:
I have added a check for an exploration being in progress.

I have opted to keep this class, the listeners determine the session state and passes it along to the controller.